File: platform-demos/C/samples/window.js
#!/usr/bin/gjs

var Gtk = imports.gi.Gtk;
Gtk.init(null, 0);

// Create window and give it a name
// You can't call it "window" as that name is a JavaScript keyword.
var sampleWindow = new Gtk.Window({type: Gtk.WindowType.TOPLEVEL});
sampleWindow.title = "Welcome to GNOME";

/* The "destroy" signal is sent out when you click the X button.
   Here, we connect that signal to the GTK+ function to close the window. */
sampleWindow.connect("destroy", function(){Gtk.main_quit()});

/* Here are a few ways we can customize our window.
   Try uncommenting them or changing their values! */
// sampleWindow.set_default_size (400,200);
// sampleWindow.set_has_resize_grip (false);
// sampleWindow.set_opacity (0.5);
// sampleWindow.maximize ();

// If the window has widgets in it, you'll want to use show_all() instead.
sampleWindow.show();

Gtk.main();

File: platform-demos/C/samples/window.py
from gi.repository import Gtk
import sys


class MyApplication(Gtk.Application):

    def do_activate(self):
        # create a Gtk Window belonging to the application itself
        window = Gtk.Window(application=self)
        # set the title
        window.set_title("Welcome to GNOME")
        # show the window
        window.show_all()

# create and run the application, exit with the value returned by
# running the program
app = MyApplication()
exit_status = app.run(sys.argv)
sys.exit(exit_status)
